src/Addons/User/AddonsUser.php

```diff
@@ -38,45 +40,110 @@ class AddonsUser implements UserInterface
      */
     private $request;
 
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    /**
+     * @var AccountsDataProvider
+     */
+    private $accountsDataProvider;
+
     public function __construct(
-        CredentialsEncryptor $encryption
+        SessionInterface $session,
+        CredentialsEncryptor $encryption,
+        AccountsDataProvider $accountsDataProvider
     ) {
         $this->encryption = $encryption;
         $this->request = Request::createFromGlobals();
+        $this->session = $session;
+        $this->accountsDataProvider = $accountsDataProvider;
     }
 
     /**
-     * {@inheritdoc}
+     * @return bool
      */
     public function isAuthenticated()
     {
-        return $this->hasCookieAuthenticated();
+        return $this->hasAccountsTokenInSession()
+            || $this->isConnectedOnPsAccounts()
+            || $this->hasCookieAuthenticated()
+            || $this->hasSessionAuthenticated();
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasAccountsTokenInSession()
+    {
+        return null !== $this->getFromSession('accounts_token');
+    }
+
+    /**
+     * @return bool
+     */
+    public function isConnectedOnPsAccounts()
+    {
+        $accountsToken = $this->accountsDataProvider->getAccountsToken();
+
+        return !empty($accountsToken);
+    }
+
+    /**
+     * @return bool
+     */
+    private function hasSessionAuthenticated()
+    {
+        return $this->getFromSession('username_addons_v2')
+            && $this->getFromSession('password_addons_v2');
     }
 
     /**
      * {@inheritdoc}
      */
     public function getCredentials($encrypted = false)
     {
+        $accountsToken = $this->getFromSession('accounts_token');
+
+        if (null !== $accountsToken) {
+            return ['accounts_token' => (string) $accountsToken];
+        }
+
+        // accounts
+        $accountsToken = $this->accountsDataProvider->getAccountsToken();
+        if (!empty($accountsToken)) {
+            return ['accounts_token' => (string) $accountsToken];
+        }
+
         return $encrypted ?
             [
-                'username' => $this->get('username_addons'),
-                'password' => $this->get('password_addons'),
+                'username' => $this->get('username_addons_v2'),
```

Comment:
The _v2 here is not a copy/paste from the v4.x of the module because we manage to handle it differently ?